Week 04/Lecture 07/Assignment 2/Q1/assignment2/src/main/java/com/example/assignment2/config/AppConfig.java

```diff
@@ -0,0 +1,9 @@
+package com.example.assignment2.config;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ComponentScan(basePackages = "com.example.assignment2")
```

Comment:
Yes, there is a case we can skip `@ComponentScan`. `@SpringBootApplication` already includes component scanning by default for the package where the main application class resides and all its sub-packages. We just need to ensure our main application class is in the base package.

Therefore, since we are using Spring Boot, we don’t need an additional `@ComponentScan` or `@Configuration` in `AppConfig` if our classes are in the right packages. Therefore, we remove `AppConfig` as it is not doing anything special.

I have revised my assignment and implemented this.